code/controllers/subsystem/speech_controller.dm

```diff
@@ -0,0 +1,43 @@
+SUBSYSTEM_DEF(speech_controller)
+	name = "Speech Controller"
+	wait = 1
+	flags = SS_TICKER
+	priority = FIRE_PRIORITY_SPEECH_CONTROLLER//has to be high priority, second in priority ONLY to SSinput
+	init_order = INIT_ORDER_SPEECH_CONTROLLER
+	runlevels = RUNLEVELS_DEFAULT | RUNLEVEL_LOBBY
+
+	///list of the form: list(client mob, message that mob is queued to say, other say arguments (if any)).
+	///this is our process queue, processed every tick.
+	var/list/queued_says_to_execute = list()
+
+///queues mob_to_queue into our process list so they say(message) near the start of the next tick
+/datum/controller/subsystem/speech_controller/proc/queue_say_for_mob(mob/mob_to_queue, message, ...)
+	if(!ismob(mob_to_queue) || mob_to_queue.gc_destroyed || !message)
+		return FALSE
+
+	if(!TICK_CHECK)
```

Comment:
This seems a bit skuffed, since the current tick limit includes maptick, and we're past that point by now. Guess it should work?

Reply:
this should skip the queue if the server isnt overloaded